### PR TITLE
zombienet: add BEEFY justifications import test

### PR DESCRIPTION
**Pull request by acatangiu**
Regression test for https://github.com/paritytech/substrate/pull/11959

Fixes https://github.com/paritytech/substrate/issues/11761

### CODE REVIEW

**Review by JurbinM — APPROVED** [review]

**Comment by pepoviola** [comment]
Hi @acatangiu, can you bump zombienet version to `v1.2.52` in your branch? That version has the fix for pause/resume.
Thanks!

File: zombienet_tests/functional/0003-beefy-and-mmr.feature [hunk 1]
```diff
@@ -37,8 +44,13 @@ validator-0: reports substrate_beefy_best_block is at least 21 within 130 second
...
 
-# TODO 1: Custom JS to test BEEFY RPCs
-
-# TODO 2: Custom JS to test MMR RPCs
+# TODO (issue #11972): Custom JS to test BEEFY RPCs
+# TODO (issue #11972): Custom JS to test MMR RPCs
 
-# TODO 3: once zombienet/#227 is implemented: pause+resume nodes to test Justifications Import
+# Check that validator-3 is still paused somewhere in 1st session.
+validator-3: reports substrate_beefy_validator_set_id is 0
```
Comment (pepoviola) [inline zombienet_tests/functional/0003-beefy-and-mmr.feature:51]: @acatangiu `pause` command will pause the polkadot process, so you can't query the metrics on that node until is `resumed`.

Comment (acatangiu) [reply zombienet_tests/functional/0003-beefy-and-mmr.feature:51]: Looking at Line 2076 in [job logs](https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/1729228), it seems that the problem is **after** resume, not before.

Comment (acatangiu) [reply zombienet_tests/functional/0003-beefy-and-mmr.feature]: @pepoviola I removed the queries on the paused node, but failure is still the same: Metrics aren't accessible for node post-resume: https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/1729228

I see the [zombienet pause/resume tests](https://github.com/paritytech/zombienet/blob/86c383053de37f8760371bd433abb89a59e43b7e/tests/0002-restart-pause-resume-node.feature#L15) don't cover metrics or any other advanced querries post-resume, so I don't think the issue is related to this PR, but rather a zombienet bug.

Comment (pepoviola) [reply zombienet_tests/functional/0003-beefy-and-mmr.feature]: Hi @acatangiu, cna you bump your fork to `v1.2.54`. I just fix the issue and check that works in your test.
```
    ✔ validator-0: is up (1888ms)
    ✔ validator-1: is up (1801ms)
    ✔ validator-2: is up (1808ms)
    ✔ validator-3: is up (1854ms)
    ✔ validator-0: reports node_roles is 4
    ✔ validator-1: reports node_roles is 4
    ✔ validator-2: reports node_roles is 4
    ✔ validator-3: reports node_roles is 4
    ✔ validator-0: reports substrate_beefy_validator_set_id is 0 (10001ms)
    ✔ validator-1: reports substrate_beefy_validator_set_id is 0 (10003ms)
    ✔ validator-2: reports substrate_beefy_validator_set_id is 0 (10001ms)
    ✔ validator-3: reports substrate_beefy_validator_set_id is 0
    ✔ validator-0: reports substrate_beefy_best_block is at least 1 within 60 seconds
    ✔ validator-1: reports substrate_beefy_best_block is at least 1 within 60 seconds
    ✔ validator-2: reports substrate_beefy_best_block is at least 1 within 60 seconds
    ✔ validator-3: reports substrate_beefy_best_block is at least 1 within 60 seconds (3870ms)
    ✔ validator-3: pause (3503ms)
    ✔ validator-0: reports substrate_beefy_validator_set_id is at least 1 within 70 seconds
    ✔ validator-1: reports substrate_beefy_validator_set_id is at least 1 within 70 seconds
    ✔ validator-2: reports substrate_beefy_validator_set_id is at least 1 within 70 seconds
    ✔ validator-0: reports substrate_beefy_validator_set_id is at least 2 within 130 seconds
    ✔ validator-1: reports substrate_beefy_validator_set_id is at least 2 within 130 seconds
    ✔ validator-2: reports substrate_beefy_validator_set_id is at least 2 within 130 seconds
    ✔ validator-0: reports substrate_beefy_best_block is at least 21 within 130 seconds (43830ms)
    ✔ validator-1: reports substrate_beefy_best_block is at least 21 within 130 seconds (1893ms)
    ✔ validator-2: reports substrate_beefy_best_block is at least 21 within 130 seconds
    ✔ validator-3: resume (3445ms)
    ✔ validator-3: reports substrate_beefy_validator_set_id is at least 2 within 30 seconds
    ✔ validator-3: reports substrate_beefy_best_block is at least 21 within 30 seconds (1823ms)
    ```
    

Thanks!!

Comment (acatangiu) [reply zombienet_tests/functional/0003-beefy-and-mmr.feature]: Thanks you @pepoviola for the quick turnaround here! :+1:

File: zombienet_tests/functional/0003-beefy-and-mmr.feature [hunk 2]
```diff
@@ -6,21 +6,28 @@ Creds: config
...
 validator-2: reports substrate_beefy_validator_set_id is 0
+validator-3: reports substrate_beefy_validator_set_id is 0
 
 # Verify voting happens and 1st mandatory block is finalized within 1st session.
 validator-0: reports substrate_beefy_best_block is at least 1 within 60 seconds
 validator-1: reports substrate_beefy_best_block is at least 1 within 60 seconds
 validator-2: reports substrate_beefy_best_block is at least 1 within 60 seconds
+validator-3: reports substrate_beefy_best_block is at least 1 within 60 seconds
+
+# Pause validator-3 and test chain is making progress without it.
```
Comment (sandreim) [inline zombienet_tests/functional/0003-beefy-and-mmr.feature:29]: Based on the comment, I would expect to check block height after resume. But we actually want to see if we catchup with all BEEFY justifications.

**Review by sandreim — APPROVED** [review]

**Review by pepoviola — APPROVED** [review]

**Review by sergejparity — APPROVED** [review]

**Comment by acatangiu** [comment]
@davxy can you please take a look here as well? It is a zombienet regression test for PR https://github.com/paritytech/substrate/pull/11959 you reviewed.

**Comment by shawntabrizi** [comment]
bot rebase

**Comment by acatangiu** [comment]
bot rebase

**Review by davxy — APPROVED** [review]

**Pull request merged**